### PR TITLE
R4R: Add gateway lcd and docs

### PR DESCRIPTION
**Pull request by secret2830**
Resolves: #1521

### CODE REVIEW

File: docs/light-client/README.md [hunk 1]
```diff
@@ -140,8 +140,14 @@ Once IRISLCD is started, you can open `localhost:1317/swagger-ui/` in your explo
     14. `GET /service/fees/{address}`:  Query service fees of a address
     15. `POST /service/fees/{address}/refund`: Refund service return fee of consumer
     16. `POST /service/fees/{address}/withdraw`: Withdraw service incoming fee of provider
-    
-10. Query app version
+
+10. Asset module APIs
+    1. `GET /asset/gateways/{moniker}/gateway`: Query the gateway of a given moniker
+    2. `GET /asset/gateways/{owner}/gateways`: Query all gateways of an owner
+    3. `POST /asset/gateways/{owner}/create`: Create a gateway
```
Comment (chengwenxi) [inline docs/light-client/README.md]: GET       /asset/gateways/{moniker}
POST    /asset/gateways

Comment (chengwenxi) [reply docs/light-client/README.md]: reference: https://restfulapi.net/resource-naming/

Comment (secret2830) [reply docs/light-client/README.md]: Done

File: client/asset/lcd/query.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		"/asset/gateways/{moniker}",
+		monikerGatewayHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+
+	// Get all gateways from an owner
+	r.HandleFunc(
+		"/asset/gateways",
+		ownerGatewaysHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+}
```
Comment (haifengxi) [inline client/asset/lcd/query.go]: Let's keep only two query endpoints: 
- `GET /asset/gateways/{moniker}` 
- `GET /asset/gateways` which takes an optional query param: `owner`; if no param is given, then return a list of all gateways.

Comment (secret2830) [reply client/asset/lcd/query.go]: Done

File: client/asset/lcd/sendtx.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		"/asset/gateways",
+		createGatewayHandlerFn(cdc, cliCtx),
+	).Methods("POST")
+
+	// edit a gateway
+	r.HandleFunc(
+		"/asset/gateways/{moniker}/edit",
+		editGatewayHandlerFn(cdc, cliCtx),
+	).Methods("POST")
+}
```
Comment (haifengxi) [inline client/asset/lcd/sendtx.go]: Edit gateway API should be `PUT /asset/gateways/{moniker}`

Comment (secret2830) [reply client/asset/lcd/sendtx.go]: Yes

File: app/v1/asset/querier.go [hunk 4]
```diff
@@ -90,7 +90,29 @@ func queryGateways(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) ([]byt
...
+		// if the owner provided
+		gateways = queryGatewaysByOwner(ctx, params.Owner, keeper)
+	} else {
+		// if the owner not given
+		gateways = queryAllGateways(ctx, keeper)
+	}
+
+	if len(gateways) == 0 {
+		return nil, nil
+	}
```
Comment (chengwenxi) [inline app/v1/asset/querier.go]: will return empty, expect `[]`.

Comment (secret2830) [reply app/v1/asset/querier.go]: Done

**Pull request merged**